src/hotspot/share/gc/g1/g1RemSet.cpp

```diff
@@ -853,13 +849,15 @@ class G1ScanCollectionSetRegionClosure : public HeapRegionClosure {
     // The individual references for the optional remembered set are per-worker, so we
     // always need to scan them.
     if (r->has_index_in_opt_cset()) {
+      EventGCPhaseParallel event;
```

Comment:
An option to avoid the per-region events (this and code roots) is to make a single event with two values in them. Just a suggestion.

Reply:
Yes, that or create a phase name that clearly includes both. I'll filed an RFE to look at this: [JDK-8297692](https://bugs.openjdk.org/browse/JDK-8297692)